resqpy/property.py

```diff
@@ -4878,22 +4878,18 @@ def validate_uom_from_string(input_unit, case_sensitive=True):
    if not input_unit:
       return default
 
-   # Get set of all valid uoms
-   valid_uoms = set(bwam.properties_data()['uoms'])
-
    if case_sensitive:
-      if input_unit in valid_uoms:
+      if input_unit in bwam.valid_uoms():
```

Comment:
Perhaps we should emit a warning if a valid uom is not found?

Reply:
Could we go further and raise an Exception if a valid unit is not found? I'm a bit worried about "failing silently".

For example, suppose a property is stored in e.g. "pore units", which is not a valid uom. This might get saved as "EuC", which can then be converted into units of "%". This will then be wrong by a factor of 100.

On a related note - I think there is some duplication between `property.validate_uom_from_string` and `bwam.rq_uom`. Do you think they should be consolidated?